### ISSUE TITLE
Rows consumed from RabbitMQ into a table do not trigger the materialized views defined on the table

### ISSUE THREAD

**Issue by chjahnert** [ClickHouse/ClickHouse#57806]
We have a `ReplicatedAggregatingMergeTree` table that is populated via a materialized view that consumes data from RabbitMQ. The table is populated as it should be, however, we have some materialized views on the base-table which are not triggered when new data is inserted from the RabbitMQ consumer. Here is the base table:

```
CREATE TABLE IF NOT EXISTS "chat_events_local" ON CLUSTER "analytics"
(
  "id" String CODEC(ZSTD),
  "customer_id" Int32 CODEC(ZSTD(8)),
  "engine_id" Int32 CODEC(ZSTD(8)),
  "event_type" Enum('question' = 0, 'answer' = 1, 'error' = 2) CODEC(T64, ZSTD(8)),
  "date" DateTime64(4, 'UTC') CODEC(Delta, ZSTD(8)),
  ...
)
ENGINE = ReplicatedMergeTree('/clickhouse/{database}/tables/{shard}/{table}', '{replica}')
PRIMARY KEY ("customer_id", "engine_id", "event_type", "date")
PARTITION BY toYYYYMM ("date");

CREATE TABLE IF NOT EXISTS "chat_events" ON CLUSTER "analytics" AS "chat_events_local"
ENGINE = Distributed("analytics", "analytics", "chat_events_local", rand());
```

We also have a set of materialized views to show aggregated data from this table:

```
CREATE TABLE IF NOT EXISTS "chat_conversations_local" ON CLUSTER "analytics"
(
  "customer_id" Int32,
  "engine_id" Int32,
  "conversation_id" UUID,
  "first" DateTime64(4, 'UTC'),
  "last" DateTime64(4, 'UTC'),
  ...
  PROJECTION "chat_conversations_by_id" (SELECT * ORDER BY "conversation_id", "engine_id", "customer_id")
)
ENGINE = ReplicatedAggregatingMergeTree('/clickhouse/{database}/tables/{shard}/{table}', '{replica}')
ORDER BY ("customer_id", "engine_id", "first", "conversation_id");
SETTINGS
  index_granularity = 8192,
  min_age_to_force_merge_seconds = 7200;

CREATE MATERIALIZED VIEW "chat_conversations_mv" ON CLUSTER "analytics"
TO
  "chat_conversations_local"
AS
SELECT
  "customer_id",
  "engine_id",
  "conversation_id",
  MIN("date") AS "first",
  MAX("date") AS "last",
  ...
FROM
  "chat_events"
GROUP BY
  "customer_id",
  "engine_id",
  "conversation_id";
```

And finally, we populate the `chat_events_local` table from data that is consumed from RabbitMQ with the following objects:

```
CREATE TABLE "chat_events_queue" ON CLUSTER "analytics" AS "chat_events"
ENGINE = RabbitMQ(rabbit1)
SETTINGS
  rabbitmq_exchange_name = 'chat-events',
  rabbitmq_routing_key_list = 'chat-events',
  rabbitmq_queue_base = 'chat-events-ch';

CREATE MATERIALIZED VIEW "chat_events_consumer" ON CLUSTER "analytics"
  TO "chat_events_local" AS
  SELECT * FROM "chat_events_queue";
```

Consuming the data from RabbitMQ works fine and the `chat_events_local` table is populated as it should be. However, when the data is consumed from RabbitMQ and it is inserted into the `chat_events_local` table, the materialized views defined on that table are not updated (e.g. `chat_conversations_mv`). The materialized views are updated correctly when using an INSERT statement, but not if the data is coming from RabbitMQ.

In other words, rows consumed from RabbitMQ into a table do not trigger the materialized views defined on that table. Only INSERTS do.

**Comment by chjahnert** [comment]
Hi there, do you have any updates on this issue?
